crates/noirc_evaluator/src/ssa/acir_gen/acir_mem.rs

```diff
@@ -123,6 +118,81 @@ impl ArrayHeap {
         self.staged.insert(index, (value, op));
     }
 
+    /// This helper function transforms an expression into a linear expression, by generating a witness if the input expression is not linear nor constant
+    fn normalize_expression(expr: &Expression, evaluator: &mut Evaluator) -> Expression {
+        if !expr.is_linear() || expr.linear_combinations.len() > 1 {
```

Comment:
```suggestion
        if !expr.is_degree_one_univariate() {
```

Reply:
Ah my bad, I missed that this omitted constants. Thoughts on flipping this if-else around to check for what we want rather than what we don't want?

```suggestion
    fn normalize_expression(expr: &Expression, evaluator: &mut Evaluator) -> Expression {
        if expr.is_const() || expr.is_degree_one_univariate()  {
          expr.clone()
        } else {
            let w = evaluator.create_intermediate_variable(expr.clone());
            Expression::from(w)
        }
    }
```

Reply:
I find it clearer to show what we do (i.e create a witness if not linear), instead of what we want (i.e a linear expression), but don't have a strong opinion either so I switched.